### DISCUSSION TITLE
About Open API spec and the Pull Request #20

### DISCUSSION THREAD

**Discussion by metalalive** [Game-as-a-Service/Pandemic-Reign-of-Cthulhu#22] [General]
![gaas-pandemic-github-cd-build-error](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/assets/7197142/01226160-ab98-460c-a6db-8eefdbdfed9e)

@sandytang886 ,
Could you describe which part you don't understand from [the build result of PR #20](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/pull/20) ?

- build error ?
  - is this your question ?
  - can you describe more about your question ? such as which parts you think it is weird,
- Open API syntax error ?
  - I checked [the API doc page](https://game-as-a-service.github.io/Pandemic-Reign-of-Cthulhu/doc/api/) few minutes later after the PR was merged, all the 3 endpoints I added are present without error
  - I also have swagger editor in my environment , to detect any syntax error beforehand.

Also
It is always good to know about Open API syntax,
Here is [the reference documents](https://github.com/OAI/OpenAPI-Specification/tree/main/versions) about different versions of Open API specification,
you can learn the basic building blocks, but no need to dive into all the detail,
it is OK to have just enough knowledge to add endpoints.
So next time when you review someone's Open API spec, you will know what to do and expect.

**Comment by metalalive** [comment]
@sandytang886 , I didn't receive your reply so far,
if you don't know how to ask question about technical issues,
it is highly recommended that you read the articles
- [How To Ask Questions The Smart Way - Eric Steven Raymond](http://www.catb.org/~esr/faqs/smart-questions.html) 
- [提問的智慧](https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way)

That would be helpful to reorganize the description of your original question.
(allow me to post your original vague question at here)

> add api doc 那邊我review 完東西好像是怪怪，可能是我review 東西沒有很懂就approval 了，doc的問題是需要再解決嗎？ 我可能也要再找一些文檔去再了解code和api doc怎麼做

**Comment by sandytang886** [comment]
抱歉比較晚回覆，真的不好意思，近日有工作累還沒有機會整理回覆，但很謝謝你的建議

.build error
對，我是說這個error有沒有影響到api doc，因為我approve 之後才發現這個問題

.open api error
這是應該是我誤解，這部份沒有問題

**Comment by metalalive** [comment]
> .build error
對，我是說這個error有沒有影響到api doc，因為我approve 之後才發現這個問題

請參考這些頁面 : 
- GitHub Action [全部 build-and-deploy 結果, 依發生時間排序 ](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/actions)
- PR20 [每次 build-and-deploy 歷史結果](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/pull/20/checks)

GitHub 什麼時候觸發 build-and-deploy process ?
- 每次有新commits 上傳到遠端 GitHub 伺服器時 e.g. `git push -u xxx your-remote-branch`
- [GitHub static page configuration](https://github.com/Game-as-a-Service/Pandemic-Reign-of-Cthulhu/settings/pages) 被修改

你之前 approve PR20 時, 我多次更改 GitHub static page 設置,
可能是我操作速度很快, 導致 Github build system 無法跟上, 然後發生錯誤
之後  build-and-deploy process 恢復正常.

**Comment by teds-lin** [comment]
我在撰寫 #13 有查閱這邊文件，目前有以下建議：
- Server那邊的下拉式選單，能否新增(或者修改成)以下伺服器
`http://127.0.0.1:8081`
方便進行測試

-  api endpoint POST `/games` 的example **input**
修改成符合遊戲規則的2~4名玩家, 而非目前的僅一位玩家

**Reply by metalalive** [reply]
OK, I'll edit Open API spec again later
